src/macros.rs

```diff
@@ -267,3 +267,35 @@ macro_rules! define_casts {
         )+
     }
 }
+
+#[macro_export]
+macro_rules! cfg_feature_enabled {
+    ($name:tt) => (
+        {
+            #[cfg(target_feature = $name)]
+            {
+                true
+            }
+            #[cfg(not(target_feature = $name))]
+            {
+                __unstable_detect_feature!($name)
+            }
+        }
+    )
+}
+
+#[cfg(not(any(target_arch = "x86",
+              target_arch = "x86_64")))]
+#[macro_export]
+macro_rules! __unstable_detect_feature {
+    ($t:tt) => { false }
```

Comment:
Is this the interface we want to end up having? If you're on x86 I think it's a static error to test for a feature that doesn't exist, so maybe it should be a static error to call this macro on other platforms with unknown features?

Reply:
I'd prefer that. I don't know the best way to do this though. I madae this the case on the x86 macro by only including matches for existing features. Should I just leave the macro empty?